services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,18 +45,84 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
-  return ERR_CODE_SUCCESS;
+  if (event == NULL)
+  {
+    return ERR_CODE_INVALID_ARG; 
+  }
+
+  if (thermalMgrQueueHandle == NULL)
+  {
+    return ERR_CODE_INVALID_STATE; 
+  }
+
+  if (xQueueSend(thermalMgrQueueHandle, (void *) event, 0) == pdTRUE)
+  {
+    return ERR_CODE_SUCCESS;
+  }  
+
+  return ERR_CODE_QUEUE_FULL; 
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t interrupt = {0};
+  interrupt.type = THERMAL_MGR_EVENT_OVER_TEMP_SHUTDOWN;
+  thermalMgrSendEvent(&interrupt);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  /* Implement this task */   
+  lm75bd_config_t lm75bdConfig = *(lm75bd_config_t *) pvParameters;
+
+
   while (1) {
-    
+    thermal_mgr_event_t event;
+
+    if (thermalMgrQueueHandle != NULL)
+    {
+    if (xQueueReceive(thermalMgrQueueHandle, &event, 5000/portTICK_PERIOD_MS) == pdPASS)
+    {
+        float currtemp;
+        if (event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD)
+        {
+          error_code_t errCode = readTempLM75BD(lm75bdConfig.devAddr, &currtemp);
+          
+          if (errCode == ERR_CODE_SUCCESS)
+          {
+            addTemperatureTelemetry(currtemp);
+          }
+
+          else if (errCode != ERR_CODE_SUCCESS)
```

Comment:
nit: this can just be an else instead of else if since if its not == ERR_CODE_SUCCESS then its != ERR_CODE_SUCCESS